src/process/personality.rs

```diff
@@ -0,0 +1,1695 @@
+use core::convert::{TryFrom, TryInto};
+use core::num::NonZeroU64;
+use core::ptr::NonNull;
+use core::sync::atomic::AtomicU8;
+use core::{mem, ptr};
+
+use bitflags::bitflags;
+
+use crate::fd::{AsRawFd, BorrowedFd};
+use crate::ffi::CString;
+use crate::process::{Pid, RawNonZeroPid, RawPid};
+use crate::{backend, io};
+
+use backend::c::{c_int, c_uint, c_void};
+use backend::process::syscalls;
+use backend::process::types::Signal;
+
+/*
+ * Helper functions.
+ */
+
+#[inline]
+fn prctl_1arg(option: c_int) -> io::Result<c_int> {
+    const NULL: *mut c_void = ptr::null_mut();
+    syscalls::prctl(option, NULL, NULL, NULL, NULL)
+}
+
+#[inline]
+fn prctl_2args(option: c_int, arg2: *mut c_void) -> io::Result<c_int> {
+    const NULL: *mut c_void = ptr::null_mut();
+    syscalls::prctl(option, arg2, NULL, NULL, NULL)
+}
+
+#[inline]
+fn prctl_3args(option: c_int, arg2: *mut c_void, arg3: *mut c_void) -> io::Result<c_int> {
+    syscalls::prctl(option, arg2, arg3, ptr::null_mut(), ptr::null_mut())
+}
+
+#[inline]
+fn prctl_get_at_arg2_optional<P>(option: i32, initial_value: P) -> io::Result<P> {
+    let mut value: P = initial_value;
+    prctl_2args(option, ((&mut value) as *mut P).cast())?;
+    Ok(value)
+}
+
+#[inline]
+fn prctl_get_at_arg2<P, T>(option: i32) -> io::Result<T>
+where
+    P: Default,
+    T: TryFrom<P, Error = io::Errno>,
+{
+    let mut value: P = Default::default();
+    prctl_2args(option, ((&mut value) as *mut P).cast())?;
+    TryFrom::try_from(value)
+}
+
+/*
+ * PR_GET_PDEATHSIG/PR_SET_PDEATHSIG
+ */
+
+const PR_GET_PDEATHSIG: c_int = 2;
+
+/// Get the current value of the parent process death signal.
+#[inline]
+pub fn parent_process_death_signal() -> io::Result<Option<Signal>> {
+    prctl_get_at_arg2_optional::<c_int>(PR_GET_PDEATHSIG, -1).map(Signal::from_raw)
+}
+
+const PR_SET_PDEATHSIG: c_int = 1;
+
+/// Set the parent-death signal of the calling process.
+#[inline]
+pub fn set_parent_process_death_signal(signal: Option<Signal>) -> io::Result<()> {
+    let signal = signal.map_or(0_usize, |signal| signal as usize);
+    prctl_2args(PR_SET_PDEATHSIG, signal as *mut _).map(|_r| ())
```

Comment:
I honestly don't like `.map(|_r| ())`. I also don't need to assign to an unused variable (you probably meant `_f` instead of `f_`) because the return value doesn't have the `#[must_use]` attribute. But what I also don't like is to add the useless line to indicate success:
```rust
Ok(())
```